src/tir/transforms/dtype_conversion.h

```diff
@@ -0,0 +1,152 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file dtype_conversion.h
+ * \brief Header file of data type conversion routines.
+ */
+#ifndef TVM_TIR_TRANSFORMS_DTYPE_CONVERSION_H_
+#define TVM_TIR_TRANSFORMS_DTYPE_CONVERSION_H_
+
+#include <tvm/tir/builtin.h>
+#include <tvm/tir/op.h>
+#include <tvm/tir/stmt.h>
+
+namespace tvm {
+
+namespace tir {
+
+/*!
+ * \brief Rounding mode: https://en.wikipedia.org/wiki/Rounding
+ */
+enum class RoundingMode {
+  // Round half to nearest even
+  kHalfToEven = 0U,
+  // Round down
+  kDown = 1U,
+  // Round up
+  kUp = 2U,
+  // Round towards zero
+  kTowardsZero = 3U,
+};
+
+/*!
+ * \brief Floating point representation.
+ */
+class FloatConfig {
+ public:
+  /*!
+   * \brief Style of infinite number representation.
+   */
+  enum class InftyStyle {
+    // Exponent all ones, mantissa all zeros
+    kIEEE = 0U,
+    // No representation of infinity
+    kNone = 1U
+  };
+  /*!
+   * \brief Style of NaN (not-a-number) representation.
+   */
+  enum class NaNStyle {
+    // Exponent all ones, mantissa non zeros
+    // - quiet NaN : 1XXXXX...
+    // - signaling NaN : 0XXXXX...
+    kIEEE = 0U,
+    // No representation of infinity
+    kNone = 1U,
+    // Both exponent bits and mantissa bits are all ones.
+    kAllOnes = 2U,
+  };
+  // The number of exponent bits.
+  int exponent;
+  // The number of mantissa (also know as fraction in IEEE format) bits.
+  int mantissa;
+  // The exponent bias in IEEE format.
+  int bias;
+  // The representation of infinity.
+  InftyStyle infty_style;
+  // The representation of NaN (Not a Number).
+  NaNStyle nan_style;
+
+  FloatConfig(int exponent, int mantissa, int bias, InftyStyle infty_style, NaNStyle nan_style)
+      : exponent(exponent),
+        mantissa(mantissa),
+        bias(bias),
+        infty_style(infty_style),
+        nan_style(nan_style) {}
+
+  inline int bits() const { return mantissa + exponent + 1; }
+
+  static FloatConfig FromDataType(DataType dtype) {
+    CHECK(dtype.is_float() || dtype.is_bfloat16() || dtype.is_float8())
+        << "FloatConfig is only applicable to floating point data types, got " << dtype
+        << " instead.";
+    if (dtype.is_float()) {
+      // IEEE 754 encoding
+      switch (dtype.bits()) {
+        case 16:
+          return FloatConfig(5, 10, 15, InftyStyle::kIEEE, NaNStyle::kIEEE);
+        case 32:
+          return FloatConfig(8, 23, 127, InftyStyle::kIEEE, NaNStyle::kIEEE);
+        default:
+          // float64
+          return FloatConfig(11, 52, 1023, InftyStyle::kIEEE, NaNStyle::kIEEE);
+      }
+    } else if (dtype.is_bfloat16()) {
+      // bfloat16
+      return FloatConfig(8, 7, 127, InftyStyle::kIEEE, NaNStyle::kIEEE);
+    } else {
+      // float8: e5m2 or e4m3
+      switch (dtype.code()) {
+        case DataType::kE4M3Float:
+          return FloatConfig(4, 3, 7, InftyStyle::kNone, NaNStyle::kAllOnes);
```

Comment:
Could you please write a comment to explain why it's not `InftyStyle::kIEEE, NaNStyle::kIEEE`?

Reply:
Because E4Me does not follow IEEE format, e.g. there is no infinity in e4m3 float8. See table 1 in [their paper](https://arxiv.org/pdf/2209.05433.pdf).

Reply:
Thanks for the explanation. Please add this comment to the code, so that everyone can see it